### ISSUE TITLE
Plugin gives null pointer exception when no dependencyManagement present

### ISSUE THREAD

**Issue by Zweater** [commsen/bom-helper-maven-plugin#3]
The plugin gives a null pointer exception when the pom.xml file does not contain a dependencyManagement section.

I am not a Java programmer but I suspect this is caused in:
https://github.com/commsen/bom-helper-maven-plugin/blob/master/src/main/java/com/commsen/maven/plugin/bomhelper/BomResolveMojo.java#L57

I suspect project.getDependencyManagement() may be NULL when there is no dependencyManagement section.
When NULL, all remaining code in the execute() function may be skipped.

Usecase for this function is to add the plugin to a generic maven parent which is by default active. It should not cause exceptions in the parent itself, nor in the childs which may not have a dependencyManagement section.

I can reproduce with a minimal pom.xml file, which I can build succesfully with

`mvn clean install`

When removing line 11 (`<dependencyManagement></dependencyManagement>`), the plugin fails with:

`Failed to execute goal com.commsen.maven:bom-helper-maven-plugin:0.3.0:resolve (check-dependencies) on project my-artifact: Execution check-dependencies of goal com.commsen.maven:bom-helper-plugin:0.3.0:resolve failed.: NullPointerException --> [Help 1]`

Best regards,

Jeroen

Minimal POM file:

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>my-test</groupId>
  <artifactId>my-artifact</artifactId>
  <version>0.0.1-SNAPSHOT</version>

  <packaging>pom</packaging>

  <dependencyManagement></dependencyManagement>

  <properties>
    <bomHelperMavenPlugin>0.3.0</bomHelperMavenPlugin>
  </properties>

  <profiles>
    <profile>
      <id>executeDependencyResolveCheck</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <build>
        <plugins>
          <plugin>
            <groupId>com.commsen.maven</groupId>
            <artifactId>bom-helper-maven-plugin</artifactId>
            <version>${bomHelperMavenPlugin}</version>
            <executions>
              <execution>
                <id>check-dependencies</id>
                <phase>verify</phase>
                <goals>
                  <goal>resolve</goal>
                </goals>
              </execution>
            </executions>
          </plugin>
        </plugins>
      </build>
    </profile>
  </profiles>
</project>
```

**Comment by Zweater** [comment]
Above reproduces with Maven 3.6.2

**Comment by azzazzel** [comment]
Thank you for reporting this @Zweater 

I'll have a look and try to fix it soon.

**Comment by azzazzel** [comment]
@Zweater, I just released the 0.4.0 version. Please check and let me know if it works for you now.

**Comment by Zweater** [comment]
@azzazzel, works like a charm!
Thank you for a great plugin!
